File: 프로그래머스/lv2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    cloth_dict = dict()
    for cloth in clothes:
        if cloth[1] in cloth_dict:
            cloth_dict[cloth[1]] += 1
        else:
            cloth_dict[cloth[1]] = 1
    for cloth in cloth_dict:
        answer *= (cloth_dict[cloth] + 1)
    answer -= 1
    
    return answer

File: 프로그래머스/lv2/42578. 의상/의상.java
import java.io.*;
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> hm = new HashMap<>();
        for (int i = 0; i < clothes.length; i++)
        {
            if (!hm.containsKey(clothes[i][1])) hm.put(clothes[i][1], 1);
            else hm.put(clothes[i][1], hm.get(clothes[i][1]) + 1);
        }
        for (String key : hm.keySet()) answer *= (hm.get(key) + 1);
        answer--;
        return answer;
    }
}